test/fuzz/lua_fuzzer/serializer.cc

```diff
@@ -0,0 +1,576 @@
+#include "serializer.h"
+
+
+CONV_FN(LuaBlock, block)
+{
+	return ChunkToString(block.chunk());
+}
+
+CONV_FN(Chunk, chunk)
+{
+	std::string ret;
+	for (int i = 0; i < chunk.stat_size(); ++i) {
+		ret += StatementWithSemicolonToString(chunk.stat(i));
+		ret += " ";
+	}
+
+	if (chunk.has_last()) {
+		ret += LastStatementWithSemicolonToString(chunk.last());
+		ret += " ";
+	}
+
+	return ret;
+}
+
+CONV_FN(StatementWithSemicolon, stat_semi)
+{
+	std::string ret = StatementToString(stat_semi.stat());
+	if (stat_semi.has_semicolon())
+		ret += "; ";
+	return ret;
+}
+
+CONV_FN(LastStatementWithSemicolon, last_stat_semi)
+{
+	std::string ret = LastStatementToString(last_stat_semi.stat());
+	if (last_stat_semi.has_semicolon())
+		ret += "; ";
+	return ret;
+}
+
+CONV_FN(Statement, stat)
+{
+	using StatType = Statement::StatOneofCase;
+	switch (stat.stat_oneof_case()) {
+	case StatType::kList:
+		return AssignmentListToString(stat.list());
+	case StatType::kCall:
+	    return FunctionCallToString(stat.call());
+	case StatType::kBlock:
+	    return DoBlockToString(stat.block());
+	// case StatType::kWhileCycle:
+	//     return WhileCycleToString(stat.whilecycle());
+	// case StatType::kRepeatCycle:
+	//     return RepeatCycleToString(stat.repeatcycle());
```

Comment:
Uncommented.

Reply:
Update: Problems with these clauses, because they can produce infinite loops and fuzzer will get stuck on them. I wrote a comment in code describing this issue.